### PR TITLE
Added behavior to ignore matching directories

### PR DESCRIPTION
**Pull request by flickyiyo**

### CODE REVIEW

**Review by jarmit — APPROVED** [review]

File: src/good_fences_runner.rs [hunk 1]
```diff
@@ -28,11 +28,12 @@ impl GoodFencesRunner {
         tsconfig_paths_json: TsconfigPathsJson,
         directory_paths_to_walk: &Vec<&str>,
         external_fences: ExternalFences,
+        ignored_dirs: Vec<String>,
```
Comment (Adjective-Object) [inline src/good_fences_runner.rs:31]: `Option<Vec<String>>`

File: good-fences.js [hunk 2]
```diff
@@ -13,13 +13,13 @@
      .option('-o, --output <string>', 'path to write found violations')
      .option('--baseUrl <string>', "Overrides `compilerOptions.baseUrl` property read from '--project' argument")
      .option('--ignoreExternalFences', 'Ignore external fences (e.g. those in `node_modules`)', false)
-     .option('--ignoredDirs', 'Ignore matching dirs (e.g. `--ignoreDirs lib` will ignore all source and fence files in all subdirs with name `lib`', [])
+     .option('--ignoredDirs [paths...]', 'Excludes matching dirs from fence evaluation (e.g. `--ignoreDirs lib` will not evaluate source files in all dirs named `lib`', [])
```
Comment (Adjective-Object) [inline good-fences.js:16]: If this epects regexes, the documentation should cover that this expects regexes

File: src/evaluate_fences.rs [hunk 3]
```diff
@@ -68,8 +68,19 @@ pub fn evaluate_fences<'fencecollectionlifetime, 'sourcefilelifetime>(
     source_files: &HashMap<String, SourceFile>,
     tsconfig_paths_json: &TsconfigPathsJson,
     source_file: &'sourcefilelifetime SourceFile,
+    ignored_dirs: Vec<String>,
```
Comment (Adjective-Object) [inline src/evaluate_fences.rs:71]: Option<Vec<String>>, this is an optional parameter. You can just `unwrap_or` and default to an empty vec.

File: src/evaluate_fences.rs [hunk 4]
```diff
@@ -68,8 +68,19 @@ pub fn evaluate_fences<'fencecollectionlifetime, 'sourcefilelifetime>(
...
 ) -> Result<Option<Vec<ImportRuleViolation<'fencecollectionlifetime, 'sourcefilelifetime>>>, String>
 {
+    match ignored_dirs.iter().find(|id| {
+        regex::Regex::new(&id.as_str())
+            .expect("unable to match --ignoredDirs")
```
Comment (Adjective-Object) [inline src/evaluate_fences.rs:76]: this error message should include the regexp that failed to compile

Comment (Adjective-Object) [reply src/evaluate_fences.rs:76]: This recompiles the regexp on every single fence. That is needlessly wasteful.

The regexps should be compiled once, and used to filter out ignored fences during walk_dirs.

**Review by Adjective-Object — CHANGES_REQUESTED** [review]

**Pull request merged**